src/main/groovy/org/moditect/gradleplugin/add/AbstractAddModuleInfoTask.groovy

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.moditect.gradleplugin.add
+
+import groovy.transform.CompileStatic
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.logging.Logger
+import org.gradle.api.logging.Logging
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.OutputDirectory
+import org.moditect.gradleplugin.ModitectExtension
+import org.moditect.gradleplugin.Util
+import org.moditect.gradleplugin.add.model.AbstractModuleConfiguration
+import org.moditect.gradleplugin.add.model.ModuleConfiguration
+import org.moditect.gradleplugin.common.ModuleId
+import org.moditect.model.GeneratedModuleInfo
+
+import static org.moditect.gradleplugin.Util.createDirectoryProperty
+
+@CompileStatic
+abstract class AbstractAddModuleInfoTask extends DefaultTask {
```

Comment:
Consider making this a `@CacheableTask`.

Reply:
It's also recommended to annotate `File` and `FileCollection` properties with `@PathSensitive` as shown at https://docs.gradle.org/current/userguide/build_cache.html#sec:task_output_caching_details

Reply:
Done.